rust/processor/src/processors/econia_processor.rs

```diff
@@ -799,8 +807,12 @@ impl ProcessorTrait for EconiaTransactionProcessor {
             for change in &info.changes {
                 match change.change.as_ref().expect("No transaction changes") {
                     Change::WriteResource(resource) => {
-                        if resource.r#type.as_ref().expect("No resource type")
-                            == &market_accounts_type
+                        let resource_type = resource.r#type.as_ref().expect("No resource type");
+                        let address = strip_hex_number(resource_type.address.to_string())?;
+                        let resource_type = format!("{address}::{}::{}", resource_type.module, resource_type.name);
+                        let address = strip_hex_number(market_accounts_type.address.to_string())?;
+                        let market_accounts_type_string = format!("{address}::{}::{}", market_accounts_type.module, market_accounts_type.name);
+                        if resource_type == market_accounts_type_string
```

Comment:
First I'll note that this algorithm disregards type arguments, which is a problem in the general case but not here.

The market accounts type address is already sanitized since it is generated via `econia_address` (which is sanitized separately), and since this operation runs for every single write resource in the txn batch, I'm concerned about efficiency

How about constructing market_accounts_type_string outside this loop?

E.g. 

```rust
        let market_accounts_type = MoveStructTag {
            address: econia_address.to_string(),
            module: "user".to_string(),
            name: "MarketAccounts".to_string(),
            generic_type_params: vec![],
        };
```

Becomes something like:

```rust
let market_accounts_type_string = format!("{econia_address}::user::MarketAccounts");
```